clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -995,7 +995,11 @@ private void handleApiVersionsResponse(List<ClientResponse> responses,
     private void handleDisconnections(List<ClientResponse> responses, long now) {
         for (Map.Entry<String, ChannelState> entry : this.selector.disconnected().entrySet()) {
             String node = entry.getKey();
-            log.info("Node {} disconnected.", node);
+            if (null != entry.getValue() && entry.getValue() == ChannelState.EXPIRED) {
+                log.debug("Idle connection to node {} disconnected.", node);
+            } else {
+                log.info("Node {} disconnected.", node);
+            }
             processDisconnection(responses, node, now, entry.getValue());
```

Comment:
The first comparison seems redundant - if the value is `null`, it won't be equal to `ChannelState.EXPIRED`. As we also use `entry.getValue()` in multiple places, I'd suggest putting it in a local variable - I personally find it more readable and it should be marginally cheaper.
```
ChannelState channelState = entry.getValue();
if (channelState == ChannelState.EXPIRED) {
    log.debug("Idle connection to node {} disconnected.", node);
} else {
    log.info("Node {} disconnected.", node);
}
processDisconnection(responses, node, now, channelState);
```
Note that this is mostly

Reply:
I though it was good hygiene to avoid NPE as much as possible, hence added the null check that should fail first and the 2nd part of the boolean expression would not be interpreted. Because the entry.getValue() was not being used in that method I thought it would be better.

Reply:
excuse me @dimitarndimitrov, I managed to confused myself, you are correct of course :)